MovingBlocks/Terasology
# Resizing issues

- Fixes the resizing issues.
- Replaced all explicit bind calls with StateChanges
- Some code refactoring to make the code more consistent across files

### How to test
Run the game, try resizing the game in windowed mode or going from fullscreen to windowed or vice versa (I've noticed an issue with the game sometimes not being able to go from windowed to fullscreen, but that's an issue outside the scope of this PR). Everything else related to rendering should be same (no weird water bugs, no screenshot crashes, no new visual artifacts).

This PR fixes the following:
#2744 
#2659
#2591
#1747
And quite possibly a few others related to resizing.
